### ISSUE TITLE
Running multiple test agents in Zonemaster::Backend needs documentation

### ISSUE THREAD

**Issue by marc-vanderwal** [zonemaster/zonemaster#1234]
I’d like to release test PR zonamester/zonemaster-backend#1121, but the testing procedure states that I need to have two test agents running in parallel.

But I can’t find anywhere in the documentation how I can run multiple test agents. Can they run on the same machine? I assume that I will get only one agent when starting the `zm-testagent` on one machine.

**Comment by matsduf** [comment]
Yes, they can run on the same server. One test agent will hit a limit, so with two you increase the number of tests that can be processed.

I think the following should be enough if they run on the same server:

1. Create a second copy of the start script.
2. Update the second copy of the start script so that there is a unique pid file and also separate log files.
3. Run both start scripts.

If you want to bind the two test agents to different queues, then a second copy of backend_config.ini must be created and different queues set to the two ini files. And the update the second start script to use the other ini file.

This should be better documented.

**Comment by pnax** [comment]
> Update the second copy of the start script so that there is a unique pid file and also separate log files.

You can pass several options to `zonemaster_backend_testagent` (see `--help`):

```
zonemaster_backend_testagent --logfile=zonemaster_testagent.log --outfile=zonemaster_testagent.out --pidfile=zonemaster_testagent.pid start
```

**Comment by marc-vanderwal** [comment]
I see; it’s rather straightforward actually.

I’ve installed Zonemaster::Backend system wide on my test VM, so the easiest option for me is to duplicate the systemd unit file. In a nutshell:

```
$ cd /path/to/zonemaster-engine-sources
$ cd share
$ sed -e 's/zm-testagent/zm-testagent-2/g' zm-testagent.service > zm-testagent-2.service
$ sudo install -v -m 644 ./zm-testagent-2.service /etc/systemd/system/
$ sudo systemctl start zm-testagent-2
```

Thanks! 👍

**Comment by matsduf** [comment]
> You can pass several options to `zonemaster_backend_testagent` (see `--help`):
> 
> ```
> zonemaster_backend_testagent --logfile=testagent.log --outfile=testagent.out --pidfile=testagent.pid start
> ```

For BSD start script you can specify another PID in rc.conf, but not log files. I will create an PR on that.
